src/components/context/stateContext.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useMemo, useState } from 'react';
+
+export const EyeClickContext = React.createContext({
+  isClicked: false,
+  setIsClicked: (isClicked: boolean) => {
+    // eslint-disable-next-line no-console
+    console.log(isClicked);
+  },
+});
+
+type PropsEyeClick = {
+  children: React.ReactNode;
+};
+
+export const EyeClickProvider: React.FC<PropsEyeClick> = ({ children }) => {
+  const [isClicked, setIsClicked] = useState(false);
+  const value = useMemo(
+    () => ({
+      isClicked,
+      setIsClicked,
+    }),
+    [isClicked],
+  );
+
+  return (
+    <EyeClickContext.Provider value={value}>
+      {children}
+    </EyeClickContext.Provider>
+  );
+};
+
+export const ShowModalContext = React.createContext({
+  isModalShowed: false,
+  setisModalShowed: (isModalShowed: boolean) => {
+    // eslint-disable-next-line no-console
+    console.log(isModalShowed);
+  },
+});
+
+type PropsShowModal = {
```

Comment:
it's redundant to use the context here, 'cause you don't have the props drilling problem in your code